### ISSUE TITLE
Can this project run on a phone?

### ISSUE THREAD

**Issue by nhoknhutnhat95** [cj-mills/Barracuda-PoseNet-Tutorial#3]
Can this project run on a phone? (android and ios)

**Comment by cj-mills** [comment]
Hi @nhoknhutnhat95 ,
You should be able to build the project for mobile. Although, another reader encountered issues when using compute shaders on Android. If you encounter the same issues, you will need to switch to using the `C Sharp Burst` worker type and likely use the CPU for the preprocessing steps. For performance reasons, I would recommend using the MobileNet model instead of the ResNet model.

**Comment by nhoknhutnhat95** [comment]
i'm trying use in android but i'm facing performance issues. i selected Mobile Net model, used c Sharp burst and unchecked use GPU.

**Comment by cj-mills** [comment]
Hi @nhoknhutnhat95 ,
Unfortunately, having to avoid using the GPU for both inference and preprocessing is basically the worst possible option from a performance perspective. This is on top of the performance limitations from using a mobile device. If you need to use Unity's native inference engine (as opposed to [PyTorch ](https://pytorch.org/mobile/android/)or [TensorFlow](https://www.tensorflow.org/lite/guide/android)), there are some things you can try.

First, try lowering the input resolution for the model (i.e. the values for `imageDims`). This can impact the accuracy of the model, but that is secondary to smooth performance.

Second, you can try [scheduled execution](https://docs.unity3d.com/Packages/com.unity.barracuda@1.0/manual/ModelExecution.html#scheduled-execution) to execute the model in pieces over a few frames ([API Documentation](https://docs.unity3d.com/Packages/com.unity.barracuda@1.0/api/Unity.Barracuda.IWorker.html)). This may or may not be acceptable depending on your use case.

Third, you can try a newer version of the Barracuda library. According to the [Barracuda Changelog](https://docs.unity3d.com/Packages/com.unity.barracuda@2.3/changelog/CHANGELOG.html), version 2.3.0 (released 2021-10-15) added a PixelShader worker (limited support currently) that allows GPU inference without requiring support for compute shaders. It also added the ability to create tensors from a RenderTexture even when you can't use compute shaders or the GPU.
